src/zenml/zen_stores/schemas/curated_visualization_schemas.py

```diff
@@ -0,0 +1,278 @@
+    CuratedVisualizationResource,
+)
+from zenml.zen_stores.schemas.base_schemas import BaseSchema
+from zenml.zen_stores.schemas.project_schemas import ProjectSchema
+from zenml.zen_stores.schemas.schema_utils import (
+    build_foreign_key_field,
+    build_index,
+)
+from zenml.zen_stores.schemas.utils import jl_arg
+
+if TYPE_CHECKING:
+    from zenml.zen_stores.schemas.artifact_schemas import ArtifactVersionSchema
+
+
+class CuratedVisualizationResourceSchema(SQLModel, table=True):
+    """Link table mapping curated visualizations to resources."""
+
+    __tablename__ = "curated_visualization_resource"
+    __table_args__ = (
+        UniqueConstraint(
+            "visualization_id",
+            name="unique_curated_visualization_resource",
+        ),
+        build_index(__tablename__, ["resource_id", "resource_type"]),
+    )
+
+    id: UUID = Field(default_factory=uuid4, primary_key=True)
+    visualization_id: UUID = build_foreign_key_field(
+        source=__tablename__,
+        target="curated_visualization",
+        source_column="visualization_id",
+        target_column="id",
+        ondelete="CASCADE",
+        nullable=False,
+    )
+    resource_id: UUID = Field(nullable=False)
+    resource_type: str = Field(nullable=False)
+
+    visualization: "CuratedVisualizationSchema" = Relationship(
+        back_populates="resource",
+    )
+
+
+class CuratedVisualizationSchema(BaseSchema, table=True):
+    """SQL Model for curated visualizations."""
+
+    __tablename__ = "curated_visualization"
+    __table_args__ = (
+        build_index(
+            __tablename__, ["artifact_version_id", "visualization_index"]
+        ),
+        build_index(__tablename__, ["display_order"]),
+    )
+
+    project_id: UUID = build_foreign_key_field(
+        source=__tablename__,
+        target=ProjectSchema.__tablename__,
+        source_column="project_id",
+        target_column="id",
+        ondelete="CASCADE",
+        nullable=False,
+    )
+    artifact_version_id: UUID = build_foreign_key_field(
+        source=__tablename__,
+        target="artifact_version",
+        source_column="artifact_version_id",
+        target_column="id",
+        ondelete="CASCADE",
+        nullable=False,
+    )
+
+    visualization_index: int = Field(nullable=False)
+    display_name: Optional[str] = Field(default=None)
+    display_order: Optional[int] = Field(default=None)
+    layout_size: str = Field(
+        default=CuratedVisualizationSize.FULL_WIDTH.value,
+        nullable=False,
+    )
+
+    artifact_version: Optional["ArtifactVersionSchema"] = Relationship(
+        sa_relationship_kwargs={"lazy": "selectin"}
+    )
+    resource: Optional[CuratedVisualizationResourceSchema] = Relationship(
+        back_populates="visualization",
+        sa_relationship_kwargs={
+            "lazy": "selectin",
+            "cascade": "all, delete",
+            "uselist": False,
+        },
+    )
```

Comment:
```suggestion
    resource: CuratedVisualizationResourceSchema = Relationship(
        back_populates="visualization",
        sa_relationship_kwargs={
            "lazy": "selectin",
            "cascade": "all, delete",
            "uselist": False,
        },
    )
```

This is also never null, right ? Because you're always creating them together and there's a cascade delete constraint involved.